json_schema/system/links.json

```
@@ -1,160 +1,249 @@
+                },
+                "link_type": {
+                    "description": "The type of this link.",
+                    "type": "string",
+                    "enum": [
+                        "process_link"
+                    ]
+                }
+            }
         },
-        "inputs": {
-          "description": "An array of inputs for this link",
-          "type": "array",
-          "items": {
+        "supplementary_file": {
             "type": "object",
-            "$ref": "#/definitions/input"
-          }
+            "required": [
+                "file_id",
+                "file_type"
+            ],
+            "properties": {
+                "file_id": {
+                    "description": "ID for this supplementary file entity",
+                    "example": "111f222a-333b-444c-555d-eee63ddfccc7",
+                    "type": "string",
+                    "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$"
+                },
+                "file_type": {
+                    "description": "concrete type of this supplementary file entity",
+                    "type": "string",
+                    "enum": [
+                        "supplementary_file"
+                    ]
+                }
+            }
         },
-        "outputs": {
-          "description": "An array of outputs for this link.",
-          "type": "array",
-          "items": {
+        "entity": {
             "type": "object",
-            "$ref": "#/definitions/output"
-          }
+            "required": [
+                "entity_id",
+                "entity_type"
+            ],
+            "properties": {
+                "entity_id": {
+                    "description": "ID of this entity",
+                    "example": "111f222a-333b-444c-555d-eee63ddfccc7",
+                    "type": "string",
+                    "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$"
+                },
+                "entity_type": {
+                    "description": "The concrete type of this entity",
+                    "type": "string",
+                    "enum": [
+                        "project"
+                    ]
+                }
+            }
         },
-        "protocols": {
-          "description": "An array of protocols for this link.",
-          "type": "array",
-          "items": {
+        "supplementary_file_link": {
             "type": "object",
-            "$ref": "#/definitions/protocol"
-          }
+            "required": [
+                "entity",
+                "supplementary_files",
+                "link_type"
+            ],
+            "properties": {
+                "entity": {
+                    "description": "An entity that refers to this supplementary file",
+                    "type": "object",
+                    "$ref": "#/definitions/entity"
+                },
+                "supplementary_files": {
```

```suggestion
                "files": {
```

Just a suggestion, to reduce the verbosity. Within a supplementary file link it should be clear that a linked file is a supplementary file.

---

LGTM!